miden-lib/asm/kernels/transaction/lib/prologue.masm

```diff
@@ -98,6 +98,92 @@ proc.process_global_inputs
     exec.memory::set_nullifier_commitment dropw
 end
 
+# KERNEL DATA
+# =================================================================================================
+
+#! Saves the hashes of the kernel procedures to memory and verifies that it matches the KERNEL_HASH
+#! public input.
+#! 
+#! Stack: [kernel_offset]
+#! Advice Stack: [KERNEL_ROOT]
+#! Advice Map: {
+#!     KERNEL_ROOT: [KERNEL_HASHES]
+#!     KERNEL_HASH: [KERNEL_PROCEDURE_HASHES]
+#! }
+#! Output: []
```

Comment:
I really like this format. Should I rework docs of all `prologue` procedures (to achieve consistency) or we can leave all other procedures as they are since they don't have such complex inputs?

Reply:
I think we can do this - but probably in a different PR.